packages/hop-node/src/cli/stake.ts

```diff
@@ -1,66 +1,137 @@
+  tx = await bridge.convertCanonicalTokenToHopToken(
+    chainSlugToId(chain)!,
+    parsedAmount
+  )
+  await tx.wait()
 }
 
-export async function staker (
-  chain: string,
-  token: string,
-  amount: number,
-  action: StakerAction
+async function stake (
+  bridge: L2Bridge | L1Bridge,
+  parsedAmount: BigNumber
 ) {
-  if (!chain) {
+  logger.debug('Staking')
+
+  const token: Token | void = await getToken(bridge)
+  const stakeTokenBalance: BigNumber = await getTokenBalance(bridge, token)
+  const formattedAmount = bridge.formatUnits(parsedAmount)
+  if (stakeTokenBalance.lt(parsedAmount)) {
     throw new Error(
-      'chain is required. Options are: ethereum, xdai, polygon, optimism, arbitrum'
+      `not enough hToken balance to stake. Have ${bridge.formatUnits(
+        stakeTokenBalance
+      )}, need ${formattedAmount}`
     )
   }
-  if (!token) {
-    throw new Error(
-      'token is required: Options are: USDC, DAI, etc... Use correct capitalization.'
+
+  let tx
+  if (token) {
+    logger.debug('Approving token stake, if needed')
+    const spender = bridge.getAddress()
+    tx = await token.approve(spender, parsedAmount)
+    await tx?.wait()
+  }
+
+  logger.debug(`Attempting to stake ${formattedAmount} tokens`)
+  tx = await bridge.stake(parsedAmount)
+  logger.info(`Stake tx: ${(tx.hash)}`)
+  const receipt = await tx.wait()
+  if (receipt.status) {
+    logger.debug(`Stake successful`)
+  } else {
+    logger.error('Stake unsuccessful')
+  }
+}
+
+async function pollConvertTxReceive (bridge: L2Bridge, convertAmount: BigNumber) {
+  const l2Bridge = bridge as L2Bridge
+  const bonderAddress = await bridge.getBonderAddress()
+  while (true) {
+    const blockNumber = await l2Bridge.getBlockNumber()
+    const start = blockNumber - 100
+    const end = blockNumber
+    const events = await l2Bridge.getTransferFromL1CompletedEvents(
+      start,
+      end
     )
+    for (const event of events) {
+      const { amount, recipient } = event.args
+      if (recipient !== bonderAddress) {
+        continue
+      }
+      if (!amount.eq(convertAmount)) {
+        continue
+      }
+      return true
+    }
+    await wait(10 * 1000)
+  }
+}
+
+async function getToken (bridge: L2Bridge | L1Bridge): Promise<Token | void> {
+  const isEthSend: boolean = bridge.l1CanonicalTokenAddress === constants.AddressZero
+  if (isEthSend) {
+    return
+  } else if (bridge.chainSlug !== Chain.Ethereum) {
+    return (bridge as L2Bridge).hToken()
+  }
+  return (bridge as L1Bridge).l1CanonicalToken()
+}
+
+async function getTokenBalance (bridge: L2Bridge | L1Bridge, token: Token | void): Promise<BigNumber> {
+  if (!token) {
+    return bridge.getEthBalance()
   }
+  return token.getBalance()
+}
 
-  const watchers = getStakeWatchers({
+async function getBridge (token: string, chain: string): Promise<L2Bridge | L1Bridge> {
+  // Arbitrary watcher since only the bridge is needed
+  const watchers = await getWatchers({
```

Comment:
Why not just get the bridge directly here?

```typescript
import contracts from 'src/contracts'

const tokenContracts = contracts.get(token, chain)
return chain === Chain.Ethereum ? tokenContracts.l1Bridge : tokenContracts.l2Bridge
```

Reply:
I was confused thinking it was referencing the bridge contract and not the bridge class. You can disregard comment